src/Xml/XmlProvider.fs

```diff
@@ -63,8 +66,13 @@ type public XmlProvider(cfg:TypeProviderConfig) as this =
         CreateFromTextReaderForSampleList = fun reader -> 
           result.Converter <@@ XmlElement.CreateList(%reader) @@> }
 
-    generateType "XML" sample sampleIsList parseSingle parseList getSpecFromSamples 
+    let result =
+        generateType "XML" sample sampleIsList parseSingle parseList getSpecFromSamples 
```

Comment:
generateType already caches inside, so this extra cache here won't really do anything